packages/ra-ui-materialui/src/field/DateField.tsx

```diff
@@ -61,7 +61,18 @@ const DateFieldImpl = <
     }
 
     const value = get(record, source);
-    if (value == null || value === '') {
+    const parsedDate = (value: any) => {
+        if (value == null || value === '') return undefined;
+
+        return value instanceof Date
+            ? value
+            : typeof value === 'string' || typeof value === 'number'
+            ? new Date(value)
+            : undefined;
+    };
+    const date = parsedDate(value);
+
+    if (typeof date === 'undefined') {
```

Comment:
Please keep the `value == null` test instead.

Reply:
Not sure if I understood this request properly: Are you asking to have BOTH `value == null` and `typeof date === 'undefined'` or to get rid if the `undefined` criteria?

The way I see it the null check is handled before (in `parsedDate` function) and the undefined check is needed to avoid the error, kinda the whole point of this PR. Please correct me if I am wrong.

Reply:
No you're right, the check is already made in parsedDate, you can keep your code as is for this part.